tests/test_generics.py

```diff
@@ -2849,3 +2852,16 @@ def test_mix_default_and_constraints() -> None:
 
         class _(BaseModel, Generic[T]):
             x: T
+
+
+def test_generic_with_not_required_in_typed_dict() -> None:
+    T = TypingExtensionsTypeVar('T')
+
+    class FooStr(TypedDict):
+        type: NotRequired[str]
+
+    class FooGeneric(TypedDict, Generic[T]):
+        type: NotRequired[T]
+
+    assert TypeAdapter(FooStr)
+    assert TypeAdapter(FooGeneric[str])
```

Comment:
Could we run some simple validation just to ensure that those cases actually work as expected?